docs/events/2023/docker-workshop.md

```diff
@@ -0,0 +1,235 @@
+ENV NODE_ENV production
+
+# Install the app dependencies
+RUN yarn install --production
+
+# Make port 3000 available to the outside
+EXPOSE 3000
+
+# Define the command to run the app
+CMD yarn start
+```
+
+The above file has a few different parts that are important to understand but in order to understand why they are so popular in industry we need to understand how docker sees Dockerfiles. 
+
+To use an analogy, if you wanted to make a cake and share how you made it with everyone you would write down the recipe. The recipe would have a list of ingredients and a list of steps to make the cake. Dockerfiles are the same, they are a list of ingredients and steps to make your application.
+
+The steps are run one by one and each step creates a new layer in your docker image. This means that if you have a step that installs a dependency and then a step that removes it, the dependency will still be in your image. This is because the steps are run one by one and each step creates a new layer.
+
+This makes it really great for caching. If the files that a step uses haven't changed then it can use the old version of the layer and there will be no difference.
+
+### `FROM`
+
+The `FROM` command is the first command in a docker file and it defines the base image that you are using. This is the image that you are building on top of and it is the first layer in your image.
+
+### `WORKDIR`
+
+The `WORKDIR` command defines the working directory for your application. This is the directory that your application will be run from and it is the directory that you will be in when you run commands like `RUN` and `CMD`.
+
+### `COPY`
+
+The `COPY` command copies files from your local machine into the docker image. This is how you get your application into the docker image.
+
+### `ENV`
+
+The `ENV` command sets environment variables in your docker image. This is how you can set things like the `NODE_ENV` variable in your docker image. This is useful for variables that shouldn't be included in your applications for security reasons. It's a way of passing secrets from your host machine (your computer) to the docker image.
+
+### `RUN`
+
+The `RUN` command runs a command in your docker image. This is how you install dependencies and run other commands in your docker image. Docker containers are most often build on some variant of linux so you can often use common linux commands if you need to.
+
+### `EXPOSE`
+
+The `EXPOSE` command exposes a port in your docker image. This is how you can access your application from outside the docker image. This is useful for things like web servers that need to be accessed from outside the docker image.
+
+This line in the container tells docker that the container has a program that is listening on port 3000 and that it should expose that port to the outside world.
+
+### `CMD`
+
+The `CMD` command defines the command that is run when you run the docker image. This is how you define the command that runs your application.
```

Comment:
Might be worth mentioning both CMD and ENTRYPOINT and mentioning the difference between them.